### ISSUE TITLE
Decimal Numbers not saving in backend despite configured as format: decimal

### ISSUE THREAD

**Issue by l-rapp** [FriendsOfTYPO3/content-blocks#271]
Given the following definition of a Number Field in my config.yaml
```
    -
        identifier: count
        description: "The final value of the counter. Decimal places are taken into account based on the 'decimal' setting"
        type: Number
        format: decimal
```

Decimal numbers cannot be saved when entered into the backend interface. Adding decimal places makes the i-warning icon show up beside the field in the interface, but on saving no error is thrown and the number is rounded to the nearest integer.

**Comment by PKuhlmay** [comment]
I can confirm this.

It worked in CB: 0.7.17 with TYPO3 12.4.21

Didn't work in CB dev-main branch in TYPO3 13.4.0

The numbers are first rounded to 2 after the comma when I leave the input field. After saving, they are rounded to the nearest integer.

**Comment by nhovratov** [comment]
@l-rapp Thanks for reporting the issue! Fixed in 1.0.3.
